clients/admin-ui/src/features/system/system-form-declaration-tab/PrivacyDeclarationForm.tsx

```diff
@@ -0,0 +1,408 @@
+/**
+ * Exports various parts of the privacy declaration form for flexibility
+ */
+
+import { Box, Button, Collapse, Flex, Spacer, Stack } from "@fidesui/react";
+import { Form, Formik, FormikHelpers } from "formik";
+import { useMemo } from "react";
+import * as Yup from "yup";
+
+import {
+  CustomCreatableSelect,
+  CustomNumberInput,
+  CustomSelect,
+  CustomSwitch,
+  CustomTextInput,
+} from "~/features/common/form/inputs";
+import { FormGuard } from "~/features/common/hooks/useIsAnyFormDirty";
+import {
+  DataCategory,
+  Dataset,
+  DataSubject,
+  DataUse,
+  LegalBasisForProcessingEnum,
+  PrivacyDeclarationResponse,
+  SpecialCategoryLegalBasisEnum,
+} from "~/types/api";
+
+import SystemFormInputGroup from "../SystemFormInputGroup";
+
+export const ValidationSchema = Yup.object().shape({
+  data_categories: Yup.array(Yup.string())
+    .min(1, "Must assign at least one data category")
+    .label("Data categories"),
+  data_use: Yup.string().required().label("Data use"),
+});
+
+export type FormValues = Omit<PrivacyDeclarationResponse, "cookies"> & {
+  cookies?: string[];
+};
+
+const defaultInitialValues: FormValues = {
+  name: "",
+  data_categories: [],
+  data_use: "",
+  data_subjects: [],
+  egress: undefined,
+  ingress: undefined,
+  features: [],
+  legal_basis_for_processing: undefined,
+  impact_assessment_location: "",
+  retention_period: "0",
+  processes_special_category_data: false,
+  special_category_legal_basis: undefined,
+  data_shared_with_third_parties: false,
+  third_parties: "",
+  shared_categories: [],
+  cookies: [],
+  id: "",
+};
+
+const transformFormValueToDeclaration = (values: FormValues) => {
+  // transform cookies from strings into object with default values
+  const transformedCookies = values.cookies
+    ? values.cookies.map((name) => ({ name, path: "/" }))
+    : undefined;
+
+  const declaration = {
+    ...values,
+    // fill in an empty string for name: https://github.com/ethyca/fideslang/issues/98
+    name: values.name ?? "",
+    special_category_legal_basis: values.processes_special_category_data
+      ? values.special_category_legal_basis
+      : undefined,
+    third_parties: values.data_shared_with_third_parties
+      ? values.third_parties
+      : undefined,
+    shared_categories: values.data_shared_with_third_parties
+      ? values.shared_categories
+      : undefined,
+    cookies: transformedCookies,
+  };
+  return declaration;
+};
+
+export interface DataProps {
+  allDataCategories: DataCategory[];
+  allDataUses: DataUse[];
+  allDataSubjects: DataSubject[];
+  allDatasets?: Dataset[];
+}
+
+export const PrivacyDeclarationFormComponents = ({
+  allDataUses,
+  allDataCategories,
+  allDataSubjects,
+  allDatasets,
+  values,
+  privacyDeclarationId,
+}: DataProps & {
+  values: FormValues;
+  privacyDeclarationId?: string;
+}) => {
+  const legalBasisForProcessingOptions = (
```

Comment:
It would be good to wrap all of these dropdown options in a `useMemo` as well. Every time this renders all of those options are being re-instantiated